src/decode.rs

```diff
@@ -4188,20 +4192,22 @@ pub(crate) unsafe fn rav1d_decode_tile_sbrow(
     t.pal_sz_uv[1] = Default::default();
     let sb128y = t.by >> 5;
     t.a = f.a.offset((col_sb128_start + tile_row * f.sb128w) as isize);
-    t.lf_mask = f.lf.mask[(sb128y * f.sb128w + col_sb128_start) as usize..].as_mut_ptr();
+    t.lf_mask = Some((sb128y * f.sb128w + col_sb128_start) as usize);
     for bx in (ts.tiling.col_start..ts.tiling.col_end).step_by(sb_step as usize) {
         t.bx = bx;
         if c.flush.load(Ordering::Acquire) != 0 {
             return Err(());
         }
-        let cdef_idx = &mut (*t.lf_mask).cdef_idx;
+        let cdef_idx = &f.lf.mask[t.lf_mask.unwrap()].cdef_idx;
         if root_bl == BlockLevel::Bl128x128 {
-            *cdef_idx = [-1; 4];
-            t.cur_sb_cdef_idx_ptr = cdef_idx.as_mut_ptr();
+            for cdef_idx in cdef_idx {
+                cdef_idx.store(-1, Ordering::Relaxed);
+            }
+            t.cur_sb_cdef_idx = Some(0);
         } else {
-            let cdef_idx = &mut cdef_idx[(((t.bx & 16) >> 4) + ((t.by & 16) >> 3)) as usize..];
-            cdef_idx[0] = -1;
-            t.cur_sb_cdef_idx_ptr = cdef_idx.as_mut_ptr();
+            let cdef_idx = &cdef_idx[(((t.bx & 16) >> 4) + ((t.by & 16) >> 3)) as usize..];
+            cdef_idx[0].store(-1, Ordering::Relaxed);
+            t.cur_sb_cdef_idx = Some((((t.bx & 16) >> 4) + ((t.by & 16) >> 3)) as usize)
```

Comment:
Could you pull this repeated `(((t.bx & 16) >> 4) + ((t.by & 16) >> 3))` expression into a variable?